src/store/slices/testing.slice.ts

```diff
@@ -0,0 +1,206 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import { fetch } from "api";
+
+import { updateProfile } from "store/slices/profile.slice";
+import { clearAccessToken } from "store/slices/session.slice";
+
+import { isAnyTaskFailure } from "pages/certification/Certification.helper";
+import { TIMELINE_CONFIG } from "compositions/Timeline/timeline.config";
+import { processTimeLineConfig, getPlannedTestingTasks, processFinishedJson } from "compositions/Timeline/components/TimelineItem/timeline.helper";
+
+import type { RootState } from "store/rootReducer";
+import type { PlanObj } from "pages/certification/Certification.helper";
+
+export interface TestingForm {
+  repoUrl: string;
+  commitHash: string;
+  name: string;
+  version: string;
+  subject?: string;
+}
+
+export interface RunStatus {
+  status: string;
+  state?: string;
+  result: any;
+  progress: any;
+  plan: any;
+}
+
+interface TestingState {
+  form: TestingForm | null;
+  creating: boolean;
+  uuid: string | null;
+  fetching: boolean;
+  timelineConfig: any;
+  plannedTestingTasks: PlanObj[];
+  coverageFile: any;
+  resultData: any;
+  runEnded: boolean;
+  runStatus: string | null;
+  runState: string | null;
+  unitTestSuccess: boolean;
+  hasFailedTasks: boolean;
+  refetchMin: number;
+  shouldFetchRunStatus: boolean;
+}
+
+const initialState: TestingState = {
+  form: null,
+  creating: false,
+  uuid: null,
+  fetching: false,
+  timelineConfig: TIMELINE_CONFIG,
+  plannedTestingTasks: [],
+  coverageFile: null,
+  resultData: null,
+  runEnded: false,
+  runStatus: null,
+  runState: null,
+  unitTestSuccess: true,
+  hasFailedTasks: false,
+  refetchMin: 5,
+  shouldFetchRunStatus: false,
+};
+
+export const createTestRun = createAsyncThunk("createTestRun", async (payload: {}, thunkApi) => {
+  try {
+    const form = (thunkApi.getState() as RootState).testing.form!;
+    const [, , , owner, repo] = form.repoUrl.split('/');
+    const profile = (thunkApi.getState() as RootState).profile.profile!;
+    const githubToken = (thunkApi.getState() as RootState).session.accessToken || undefined;
+    await thunkApi.dispatch(updateProfile({
+      ...profile, dapp: {
+        ...profile.dapp,
+        name: form.name,
+        subject: form.subject,
+        owner, repo, githubToken
+      }
+    }));
+    await thunkApi.dispatch(clearAccessToken({}));
+
+    const response = await fetch<string>(thunkApi, {
+      method: 'POST', url: '/run', data: form.commitHash,
+      headers: {'Content-type': 'text/plain;charset=utf-8'}
```

Comment:
This is the text/plain content-type that has been referred to above. Suggest to move to it an axios instance, for better maintenance

Reply:
Ok, now this is handled by a parameter